### DISCUSSION TITLE
Template database

### DISCUSSION THREAD

**Discussion by SDMProjects** [pgmodeler/pgmodeler#1611] [General]
Pgmodeler is a very useful software. II would like to thank you all for your works.

Is it possible to create template database with Pgmodeler ?
ALTER DATABASE template_db_name WITH is_template TRUE;

Is it possible to create a database model from a template ?
CREATE DATABASE new_db_name TEMPLATE template_db_name;

Sincerely yours

**Comment by rkhaotix** [comment]
Hi!

> Is it possible to create template database with Pgmodeler ?
ALTER DATABASE template_db_name WITH is_template TRUE;

Yes. You can achieve this in two ways. First, via SQL tool by running the command directly in a database server. Second, via diff tool, where you have to set the flag "is template" in your database model and then run the diff against a live database.

> Is it possible to create a database model from a template ?
CREATE DATABASE new_db_name TEMPLATE template_db_name;

Yes. Just check "is template" in the database properties.

https://www.pgmodeler.io/support/docs/53-compare-model-and-database?v=0.9.3
https://www.pgmodeler.io/support/docs/31-database?v=0.9.3
